api/diagnostics.go

```diff
@@ -226,7 +208,14 @@ func (j *DiagnosticsJob) runBackgroundJob(nodes []dcos.Node) {
 	// place a summaryErrorsReport.txt in a zip archive which should provide info what failed during the logs collection.
 	summaryErrorsReport := new(bytes.Buffer)
 
-	j.collectDataFromNodes(nodes, summaryReport, summaryErrorsReport, zipWriter)
+	err = j.collectDataFromNodes(ctx, nodes, summaryReport, summaryErrorsReport, zipWriter)
+	j.setJobProgressPercentage(100)
```

Comment:
I'd move this into a defer in `collectDataFromNodes` since it's the primary setter of the percentages, AFAICT.

Reply:
Hmm, I've just take it out but maybe you're right